src/ucp/core/ucp_mm.c

```diff
@@ -112,13 +115,22 @@ ucs_status_t ucp_mem_rereg_mds(ucp_context_h context, ucp_md_map_t reg_md_map,
                 continue;
             }
 
+            base_address = address;
+            reg_length   = length;
+
+            if (ucp_memory_check_reg_whole_alloc(context, mem_type)) {
```

Comment:
instead of string comparison, can we build memtype mask instead of array of strings?

Reply:
@bureddy So allow the user to pass a comma separated list but parse that into a bitmask? I can do that when processing the env var but it would still need a string array. Or did you mean that the user passes a bitmask?

Reply:
i mean parse the comma-separated list into bitmask.